### PR TITLE
Detach on failed fs mount. CAS-429

### PR DESCRIPTION
**Pull request by blaisedias**
To stage a file system volume we first attach the nexus,
then do a file system mount.
Now if the file system mount fails, detach the nexus.

### CODE REVIEW

**Review by tiagolobocastro — COMMENTED** [review]
IMHO I think this whole code would benefit from a bit of refactoring, moving the different operations we perform in stages so we can undo them in case of error. 
For example, in this case, if `get_access_type` fails we might want to clean up as well.
Might also improve readability a bit as well.

**Comment by blaisedias** [comment]
Well spotted. I fixed by adding a check for access type before performing any operations. Better to check and not to do, then undo.

File: csi/src/node.rs [hunk 1]
```diff
@@ -364,6 +390,15 @@ impl node_server::Node for Node {
             ));
         };
 
+        if let Err(error) = get_access_type(&msg.volume_capability) {
```
[the remark this answers is not in the document]
Comment (blaisedias) [reply csi/src/node.rs]: Done. https://github.com/openebs/Mayastor/pull/397/commits/59b566a341068364eca83547d504e239c4c779c0

**Review by tiagolobocastro — APPROVED** [review]

**Comment by blaisedias** [comment]
Doh, rebased on master!

**Pull request closed without merging**